subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -100,6 +101,7 @@ private TestingFramework(Frameworks framework, String version) {
     private final JvmComponentDependencies dependencies;
     private boolean attachedDependencies;
     private final Action<Void> attachDependencyAction;
+    private final Property<String> testType = getObjectFactory().property(String.class).convention(TestType.UNIT_TESTS);
```

Comment:
This convention needs to be set in the plugin. I think this also lets you get rid of the field and getter too.